### ISSUE TITLE
parse.js:33:27 typeError: this is not a typed array

### ISSUE THREAD

**Issue by BHare1985** [mapbox/node-pre-gyp#406]
node v4.4.6 on Amazon Beanstalk 

I have a locked dependency on node-mapnik 3.15.13 (because its production)
I also lock dependency on grainstore 1.2.0, which eventually allows gdal 0.9.7 and that fails with node-pre-gyp. I am not sure if its a bug in gdal or node-pre-gyp so cross post: https://github.com/naturalatlas/node-gdal/issues/239

I haven't needed to reinstall since early june and now when Amazon beanstalk is attempting to reinstall it throws this error

  ```
> gdal@0.9.7 install /tmp/deployment/application/node_modules/windwalker/node_modules/grainstore/node_modules/millstone/node_modules/srs/node_modules/gdal
  > node-pre-gyp install --fallback-to-build
  
  node-pre-gyp ERR! UNCAUGHT EXCEPTION 
  node-pre-gyp ERR! stack TypeError: this is not a typed array.
  node-pre-gyp ERR! stack     at Function.from (native)
  node-pre-gyp ERR! stack     at Object.<anonymous> (/tmp/deployment/application/node_modules/windwalker/node_modules/grainstore/node_modules/millstone/node_modules/srs/node_modules/gdal/node_modules/node-pre-gyp/node_modules/tar/lib/parse.js:33:27)
  node-pre-gyp ERR! stack     at Module._compile (module.js:409:26)
  node-pre-gyp ERR! stack     at Object.Module._extensions..js (module.js:416:10)
  node-pre-gyp ERR! stack     at Module.load (module.js:343:32)
  node-pre-gyp ERR! stack     at Function.Module._load (module.js:300:12)
  node-pre-gyp ERR! stack     at Module.require (module.js:353:17)
  node-pre-gyp ERR! stack     at require (internal/module.js:12:17)
  node-pre-gyp ERR! stack     at Object.<anonymous> (/tmp/deployment/application/node_modules/windwalker/node_modules/grainstore/node_modules/millstone/node_modules/srs/node_modules/gdal/node_modules/node-pre-gyp/node_modules/tar/lib/list.js:10:16)
  node-pre-gyp ERR! stack     at Module._compile (module.js:409:26)
  node-pre-gyp ERR! System Linux 4.4.14-24.50.amzn1.x86_64
  node-pre-gyp ERR! command "/opt/elasticbeanstalk/node-install/node-v4.4.6-linux-x64/bin/node" "/tmp/deployment/application/node_modules/windwalker/node_modules/grainstore/node_modules/millstone/node_modules/srs/node_modules/gdal/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
  node-pre-gyp ERR! cwd /tmp/deployment/application/node_modules/windwalker/node_modules/grainstore/node_modules/millstone/node_modules/srs/node_modules/gdal
  node-pre-gyp ERR! node -v v4.4.6
  node-pre-gyp ERR! node-pre-gyp -v v0.10.2
  node-pre-gyp ERR! This is a bug in `node-pre-gyp`.
  node-pre-gyp ERR! Try to update node-pre-gyp and file an issue if it does not help:
  node-pre-gyp ERR!     <https://github.com/mapbox/node-pre-gyp/issues>
  npm ERR! Linux 4.4.14-24.50.amzn1.x86_64
  npm ERR! argv "/opt/elasticbeanstalk/node-install/node-v4.4.6-linux-x64/bin/node" "/opt/elasticbeanstalk/node-install/node-v4.4.6-linux-x64/bin/npm" "--production" "install"
  npm ERR! node v4.4.6
  npm ERR! npm  v2.15.5
  npm ERR! code ELIFECYCLE
  
  npm ERR! gdal@0.9.7 install: `node-pre-gyp install --fallback-to-build`
  npm ERR! Exit status 7
  npm ERR! 
  npm ERR! Failed at the gdal@0.9.7 install script 'node-pre-gyp install --fallback-to-build'.
  npm ERR! This is most likely a problem with the gdal package,
  npm ERR! not with npm itself.
  npm ERR! Tell the author that this fails on your system:
  npm ERR!     node-pre-gyp install --fallback-to-build
  npm ERR! You can get information on how to open an issue for this project with:
  npm ERR!     npm bugs gdal
  npm ERR! Or if that isn't available, you can get their info via:
  npm ERR! 
  npm ERR!     npm owner ls gdal
  npm ERR! There is likely additional logging output above.

```

**Comment by BHare1985** [comment]
It looks like Buffer.from is a node v6 feature and on Feb 22  node-tar committed a change to go from new buffer to Buffer.from, dropping all support before v6: https://github.com/npm/node-tar/commits/master/lib/parse.js

I am not sure why I only saw the problem now, and not back when i reinstalled in june.

**Comment by BHare1985** [comment]
I resolved the issue by forcing tar to version 4.4.0. node-pre-gyp should consider locking the tar version to prevent mistakes like this in the future. I made an issue in the node-tar github.
